lib/libvarnish/vav.c

```diff
@@ -474,18 +474,17 @@ int
 main(int argc, char **argv)
 {
 	const struct test_case **tc;
-	int ret = 0;
+	int ret = 0, res = 0;
 
 	(void)argc;
 	(void)argv;
 
-	for (tc = tests; ret == 0 && *tc != NULL; tc++) {
-		argv = test_run(*tc, &ret);
+	for (tc = tests; *tc != NULL; res = 0, tc++) {
+		argv = test_run(*tc, &res);
 		VAV_Free(argv);
-		if(ret)
-			return (1);
+		if(res)
+			ret++;
```

Comment:
Leave `ret` as it is to avoid noise, and add a `fail` variable instead. Assign it 1 instead of incrementing it, we may other wise hit "magic" exit statuses from the test toolchain.

Reply:
fixed